engine/src/windowing/window.cpp

```diff
@@ -12,8 +12,8 @@
 namespace gen
 {
 
-	Window::Window(int width, int height, std::string title)
-		: m_width{width}, m_height{height}, m_title{std::move(title)} // NOLINT(cppcoreguidelines-pro-type-member-init)
+	Window::Window(int width, int height, const std::string& title)
```

Comment:
Using `std::string` here requires the user to allocate one even if they are passing a literal / compile time constant. It's best to just take `char const*` and pass it through unchanged.